### PR TITLE
Image: Add missing error printing to format specific functions

### PR DESCRIPTION
**Pull request by ghost**
Plus updated docs

Note: it doesn't affect load_from_file because it uses dynamic loaders while these functions are wrappers around direct calls to functions defined inside modules without using resource loader api

Note: png functions weren't updated because png is a driver (not a module) and can't be disabled

### CODE REVIEW

**Review by radiantgurl — APPROVED** [review]
Looking good

File: core/io/image.cpp
Comment (Calinou) [inline core/io/image.cpp:2637]: Note for reviewers: OpenEXR saving is only available in editor builds. Making it available in export templates has to be handled separately, e.g. using https://github.com/godotengine/godot/pull/73003.

File: core/io/image.cpp
Comment (Calinou) [inline core/io/image.cpp:2653]:
```suggestion
			"The WebP module isn't enabled. Recompile the Godot editor or export template binary with the `module_webp_enabled=yes` SCons option.");
```

File: core/io/image.cpp
Comment (Calinou) [inline core/io/image.cpp:4031]:
```suggestion
			"The JPEG module isn't enabled. Recompile the Godot editor or export template binary with the `module_jpg_enabled=yes` SCons option.");
```

File: core/io/image.cpp
Comment (Calinou) [inline core/io/image.cpp:4039]:
```suggestion
			"The WebP module isn't enabled. Recompile the Godot editor or export template binary with the `module_webp_enabled=yes` SCons option.");
```

File: doc/classes/Image.xml
Comment (Calinou) [inline doc/classes/Image.xml:360]:
```suggestion
				[b]Note:[/b] This method is only available in engine builds with the JPG module enabled. By default, the JPEG module is enabled, but it can be disabled at build-time using the [code]module_jpg_enabled=no[/code] SCons option.
```

File: doc/classes/Image.xml
Comment (Calinou) [inline doc/classes/Image.xml:411]:
```suggestion
				[b]Note:[/b] This method is only available in engine builds with the WEBP module enabled. By default, the WebP module is enabled, but it can be disabled at build-time using the [code]module_webp_enabled=no[/code] SCons option.
```

File: doc/classes/Image.xml
Comment (Calinou) [inline doc/classes/Image.xml:488]:
```suggestion
				[b]Note:[/b] This method is only available in engine builds with the JPG module enabled. By default, the JPEG module is enabled, but it can be disabled at build-time using the [code]module_jpg_enabled=no[/code] SCons option.
```

File: doc/classes/Image.xml
Comment (Calinou) [inline doc/classes/Image.xml:521]:
```suggestion
				[b]Note:[/b] This method is only available in engine builds with the WEBP module enabled. By default, the WebP module is enabled, but it can be disabled at build-time using the [code]module_webp_enabled=no[/code] SCons option.
```

File: doc/classes/Image.xml
Comment (Calinou) [inline doc/classes/Image.xml:531]:
```suggestion
				[b]Note:[/b] This method is only available in engine builds with the WEBP module enabled. By default, the WebP module is enabled, but it can be disabled at build-time using the [code]module_webp_enabled=no[/code] SCons option.
```

**Review by Calinou — COMMENTED** [review]
Should be good to merge after applying suggestions.

File: core/io/image.cpp
Comment (dustdfg) [reply core/io/image.cpp:2637]: Good idea but I think maybe as if tools aren't enabled there is no sense to check if `save_exr_func` is null as we already know it will be null...

Very good suggestion but I will replace it with `ERR_FAIL_V_MSG`

**Comment by dustdfg** [comment]
@Calinou ready for approve?

**Comment by dustdfg** [comment]
I am not going to do anything to update this PR. But it is not bad in IMO so I am not closing it... There are people with "write access" so if you need it you can correct it. I am abandoning this PR... If you want close it

**Comment by Calinou** [comment]
- Superseded by https://github.com/godotengine/godot/pull/99758. Thanks for the contribution nonetheless!

**Comment by Mickeon** [comment]
I assume you wanted to close this one 😅

**Pull request closed without merging**
